packages/manager/src/managers/sliceTemplateLibrary/SliceTemplateLibraryManager.ts

```diff
@@ -0,0 +1,179 @@
+	async read(
+		args: SliceTemplateLibraryReadData,
+	): Promise<SliceTemplateLibraryManagerReadReturnType> {
+		assertPluginsInitialized(this.sliceMachinePluginRunner);
+
+		const isTypeScriptProject = await this.project.checkIsTypeScript();
+		const hookResult = await this.sliceMachinePluginRunner.callHook(
+			"slice-template-library:read",
+			{ ...args, isTypeScriptProject },
+		);
+
+		const { data, errors } = decodeHookResult(readHookCodec, hookResult);
+
+		return {
+			templates: data.flat()[0].templates,
+			errors,
+		};
+	}
+	async create(
+		args: SliceTemplateLibraryCreateData,
+	): Promise<SliceTemplateLibraryManagerCreateReturnType> {
+		assertPluginsInitialized(this.sliceMachinePluginRunner);
+
+		// Reading all available slice templates
+		const isTypeScriptProject = await this.project.checkIsTypeScript();
+		const hookReadResult = await this.sliceMachinePluginRunner.callHook(
+			"slice-template-library:read",
+			{ ...args, isTypeScriptProject },
+		);
+		const { data: readData, errors: readErrors } = decodeHookResult(
+			readHookCodec,
+			hookReadResult,
+		);
+		if (readErrors.length > 0) {
+			return {
+				errors: readErrors,
+			};
+		}
+
+		// Extract all existing slice IDs into an array
+		const { libraries } = await this.slices.readAllSliceLibraries();
+		const existingIds: string[] = [];
+		libraries.forEach((lib) => {
+			if (lib.sliceIDs) {
+				existingIds.push(...lib.sliceIDs);
+			}
+		});
+
+		// Create a function to get the next available ID based on the baseId
+		const getNextAvailableSlice = (baseId: string) => {
+			let nextId = baseId;
+			let counter = 1;
+			while (existingIds.includes(nextId)) {
+				counter++;
+				nextId = `${baseId}_${counter}`;
+			}
+			existingIds.push(nextId);
+
+			return { id: nextId, counter };
+		};
+
+		// Extract the slices to create from the template ids given in args
+		const slicesToCreate = readData.flat()[0].templates.map((template) => {
+			const { id, counter } = getNextAvailableSlice(template.model.id);
+
+			return {
+				...template,
+				model: {
+					...template.model,
+					id,
+					name:
+						counter > 1 ? template.model.name + counter : template.model.name,
+				},
+			};
+		});
+
+		// Initiate the slice creation process for all slices
+		const creationPromises = slicesToCreate.map((slice) => {
+			assertPluginsInitialized(this.sliceMachinePluginRunner);
+
+			return this.slices.createSlice({
+				libraryID: libraries[0].libraryID,
+				model: slice.model,
+				componentContents: slice.componentContents,
+			});
+		});
+
+		// Wait for all slices to be created
+		const creationResults = await Promise.all(creationPromises);
+
+		const mocksPromises = slicesToCreate.map((slice) => {
+			assertPluginsInitialized(this.sliceMachinePluginRunner);
+
+			return this.slices.updateSliceMocks({
+				libraryID: libraries[0].libraryID,
+				sliceID: slice.model.id,
+				mocks: slice.mocks,
+			});
+		});
+
+		const mocksResults = await Promise.all(mocksPromises);
```

Comment:
We could wait both mocks and screenshots at the same time, right?

Reply:
we could wait everything at the same time, including the first contents file. In a sense, that's what the `create` function does :) But screenshots is an array of array of promises and I think it's clearer to keep it separate